### DISCUSSION TITLE
Useful links for research

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#135] [Ideas]
### CONTENT TRANSFERRED TO https://github.com/edubart/nelua-lang/wiki/Useful-Links-for-Research

These are a couple of videos I have watched and helped me a lot to understand, for example how Lua works behind the scenes.

Clearly there are other videos in the list, such as for C++ and so forth.

| Name | URL |
|----------|-------|
| CppCon 2014: Mike Acton "Data-Oriented Design and C++" | https://www.youtube.com/watch?v=rX0ItVEVjHc |
| Garbage Collection in Lua, Roberto Ierusalimschy, PUC-Rio: LiM'19 talk 7 | https://www.youtube.com/watch?v=wGizKsOJQuE |
| Functions in Lua, Roberto Ierusalimshy, PUC-Rio: LIM'17 talk 1 | https://www.youtube.com/watch?v=wdRGOE1N-FA |
| How much does it cost - Roberto Ierusalimschy | https://www.youtube.com/watch?v=EUvgoxBm7uc |

Various articles, research papers, etc (for PDF files, please **right-click on them and save them locally**):

- [Memory Allocation Strategies](https://www.gingerbill.org/series/memory-allocation-strategies/) by GingerBill
- [C++ Reference](https://en.cppreference.com/w/) - Ideal for both C and C++ reference
- [Concurrent Cycle Collection in Reference Counted Systems (PDF)](https://researcher.watson.ibm.com/researcher/files/us-bacon/Bacon01Concurrent.pdf)
- [Compiler Design in C (PDF)](https://holub.com/goodies/compiler/compilerDesignInC.pdf)
- [Centralized Memory Management](https://sasluca.github.io/cmm)
- [Handles are the better pointers](https://floooh.github.io/2018/06/17/handles-vs-pointers.html)
- [Modern C for C++ Peeps](https://floooh.github.io/2019/09/27/modern-c-for-cpp-peeps.html)

**Comment by edubart** [comment]
Nice, anything from Roberto (the Lua creator) is usually good material, many talks from him is an inspiration for Nelua.

**Comment by stefanos82** [comment]
I will be updating it with more content every time I find something interesting.

Can you check how to pin a discussion ticket?

**Comment by edubart** [comment]
You could add to the list:

https://floooh.github.io/2018/06/17/handles-vs-pointers.html
https://floooh.github.io/2019/09/27/modern-c-for-cpp-peeps.html (some ideas applies to Nelua)
https://sasluca.github.io/cmm

And the PDFs I think are too dense for this list.

**Reply by stefanos82** [reply]
> You could add to the list:
> 
> https://floooh.github.io/2018/06/17/handles-vs-pointers.html
> https://floooh.github.io/2019/09/27/modern-c-for-cpp-peeps.html (some ideas applies to Nelua)
> https://sasluca.github.io/cmm
> 
> And the PDFs I think are too dense for this list.

Done.

**Comment by stefanos82** [comment]
Would you prefer to introduce a `useful-links` markdown file do we can add or update on PR?
